server/src/services/media.service.spec.ts

```diff
@@ -438,233 +482,189 @@ describe(MediaService.name, () => {
       );
     });
 
-    it('should run successfully', async () => {
-      assetMock.getByIds.mockResolvedValue([assetStub.image]);
-      await sut.handleGeneratePreview({ id: assetStub.image.id });
-    });
-  });
-
-  describe('handleGenerateThumbnail', () => {
-    it('should skip thumbnail generation if asset not found', async () => {
-      assetMock.getByIds.mockResolvedValue([]);
-      await sut.handleGenerateThumbnail({ id: assetStub.image.id });
-      expect(mediaMock.generateThumbnail).not.toHaveBeenCalled();
-      expect(assetMock.update).not.toHaveBeenCalledWith();
-    });
+    it.each(cartesian(Object.values(ImageFormat) as string[], Object.values(AssetFileType) as string[]))(
+      'should generate a %s %s for an image when specified',
+      async ([format, type]) => {
+        systemMock.get.mockResolvedValue({ image: { [type]: { format: format as ImageFormat } } });
+        assetMock.getById.mockResolvedValue(assetStub.image);
+        const thumbhashBuffer = Buffer.from('a thumbhash', 'utf8');
+        mediaMock.generateThumbhash.mockResolvedValue(thumbhashBuffer);
 
-    it('should skip invisible assets', async () => {
-      assetMock.getByIds.mockResolvedValue([assetStub.livePhotoMotionAsset]);
+        const previewFormat = type === AssetFileType.PREVIEW ? format : ImageFormat.JPEG;
+        const thumbnailFormat = type === AssetFileType.THUMBNAIL ? format : ImageFormat.WEBP;
+        const previewPath = `upload/thumbs/user-id/as/se/asset-id-preview.${previewFormat}`;
+        const thumbnailPath = `upload/thumbs/user-id/as/se/asset-id-thumbnail.${thumbnailFormat}`;
 
```

Comment:
This test has little value imo. Looking at it, it is not clear what it is doing. It takes reading and understanding the _test_ code and checking the different branching going on here. Tests should not have branching and should be very simple. I would prefer 4 individual, trivial tests that are easy to read.

Reply:
That's fair! The reason I did it like this is mainly so it can handle more formats or image types being added in the future. I can get rid of the test branching by splitting it into two jobs for previews and thumbnails.

Reply:
The goal of the unit test should be to cover the branches in the code. And there should ideally be a test for each branch. We don't have to use every format if it doesn't run different code paths.